disasm/disasm.cc

```diff
@@ -1822,6 +1822,15 @@ disassembler_t::disassembler_t(int xlen)
     DEFINE_RTYPE(pktt32);
   }
 
+    // ext-cmo
+  DISASM_INSN("cbo.clean", cbo_clean, 0, {&xrs1});
+  DISASM_INSN("cbo.flush", cbo_flush, 0, {&xrs1});
+  DISASM_INSN("cbo.inval", cbo_inval, 0, {&xrs1});
+  DISASM_INSN("cbo.zero", cbo_zero, 0, {&xrs1});
+  DISASM_INSN("prefetch.i", prefetch_i, 0, {&store_address});
+  DISASM_INSN("prefetch.r", prefetch_i, 0, {&store_address});
+  DISASM_INSN("prefetch.w", prefetch_i, 0, {&store_address});
```

Comment:
Disassembly support for `prefetch.[irw]` instructions must be placed before `ori` instruction because prefetch hint instructions are special encodings of `ori` instruction.

Reply:
You can test this file (`spike --isa=rv64gc_zicbom_zicbop_zicboz -l test-zicbo.bare.elf`): [test-zicbo.bare.zip](https://github.com/riscv-software-src/riscv-isa-sim/files/7740384/test-zicbo.bare.zip)

This is built with GNU Binutils + [my CBO support patchset](https://sourceware.org/pipermail/binutils/2021-December/118909.html) and my toy builder (will be open-sourced later).

```asm
    .globl  _start
    .section .entry, "ax", %progbits
    .attribute arch, "rv64gc_zicbom_zicbop_zicboz"
_start:
    prefetch.i 0(x0)
    prefetch.r 32(x1)
    prefetch.w 64(x2)
    cbo.clean x1
    cbo.clean x30
    cbo.inval x1
    cbo.inval x30
    cbo.flush x1
    cbo.flush x30
    cbo.zero x1
    cbo.zero x30
    ret
```

Reply:
That's true. I didn't realize this question before jrtc27 comment in https://github.com/riscv/sail-riscv/pull/137. It's easy to fix the problem in disasm. However, It seems hard to fix the function code of prefetch.* .  Is there any suggestion?